src/KitLoong/MigrationsGenerator/MigrateGenerateCommand.php

```diff
@@ -123,11 +135,107 @@ public function handle()
 
         $this->askIfLogMigrationTable();
 
+        $this->single = $this->option('single');
+
         $this->generateMigrationFiles($tables);
 
+        if ($this->single) {
+            $this->mergeMigrationsIntoSingle();
+        }
+
         $this->info("\nFinished!\n");
     }
 
+    /**
+     * Merge all generated migrations into a single file
+     */
+    protected function mergeMigrationsIntoSingle()
+    {
+        $creates = [];
+        $keys = [];
+
+        foreach ($this->single_creates as $table => $file) {
+            $code = file_get_contents("database/migrations/${file}.php");
+            list($up, $down) = $this->getUpDown($code);
+            $creates[$table] = [ 'up' => $up, 'down' => $down, ];
+            unlink("database/migrations/${file}.php");
+        }
+
+        foreach ($this->single_keys as $table => $file) {
+            $code = file_get_contents("database/migrations/${file}.php");
+            list($up, $down) = $this->getUpDown($code);
+            $keys[$table] = [ 'up' => $up, 'down' => $down, ];
+            unlink("database/migrations/${file}.php");
+        }
+
+        $pre = '
+<?php
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateInitialTablesAndKeys extends Migration
```

Comment:
Can we move this into template file?

Reply:
See https://github.com/kitloong/laravel-migrations-generator/tree/4.x/src/Way/Generators/templates

Reply:
Is there a sample of how to use these anywhere? otherwise I can do some ghetto regex.